### PR TITLE
add test skeleton for feature `Automatically upgrade volumes' engine to the default engine image after upgrading Longhorn`

### PR DESCRIPTION
**Pull request by PhanLe1010**
longhorn/longhorn#2152

### CODE REVIEW

**Review by shuo-wu — CHANGES_REQUESTED** [review]
We can add more tests for the setting:
1. The engine upgrade API is disabled when the setting is enabled
2. The upgrade shouldn't start when the volume is degraded.
3. Offline upgrade (you can create more volumes but do not attach them in case1)

**Comment by PhanLe1010** [comment]
> The engine upgrade API is disabled when the setting is enabled

We test it here https://github.com/longhorn/longhorn-tests/blob/b0d73df12fce9633112aa26d8c1b0c1b158158e4/manager/integration/tests/test_engine_upgrade.py#L724

> The upgrade shouldn't start when the volume is degraded.

I think it is ok to upgrade the degraded volumes

> Offline upgrade (you can create more volumes but do not attach them in case1)

We already has 5 detached volumes in case 1, do we need to create more?

**Comment by shuo-wu** [comment]
Oh sorry for missing some steps. No, 5 detached volumes are enough.

File: manager/integration/tests/test_engine_upgrade.py [hunk 1]
```diff
@@ -691,3 +691,76 @@ def test_engine_live_upgrade_with_intensive_data_writing(client, core_api, volum
...
+    5. Set concurrent-automatic-engine-upgrade-per-node-limit setting to 3
+    6. In a retry loop, verify that the number of volumes who
+       is upgrading engine is always smaller or equal to 3
+    7. Wait until the upgrades are completed (volumes' engine image changed,
+       replica mode change to RW for attached volumes, reference count of the
+       new engine image changed, all engine and replicas' engine image changed,
+       etc ...)
+    8. verify the volumes' data
+
+    Case 2: Dr volume
```
Comment (shuo-wu) [inline manager/integration/tests/test_engine_upgrade.py:721]: `DR volume`

**Review by shuo-wu — DISMISSED** [review]
In general LGTM

File: manager/integration/tests/test_engine_upgrade.py (hunk 1 above)
Comment (PhanLe1010) [reply manager/integration/tests/test_engine_upgrade.py]: Thank you. Fixed

**Review by shuo-wu — APPROVED** [review]

**Pull request merged**